src/wrapper.rs

```diff
@@ -145,12 +145,12 @@ impl dtrace_hdl {
     ///
     /// Returns `Ok(())` if the option was set successfully, or an error code if the option could
     /// not be set.
-    pub fn dtrace_setopt(&self, option: &str, value: &str) -> Result<(), Error> {
+    pub fn dtrace_setopt(self, option: &str, value: &str) -> Result<Self, Error> {
```

Comment:
My only concern is there are some options that can be set at runtime. Although we're not setting any runtime options for now but if we decide to in the future this may make things tedious.